src/app/core/services/variant-parser.service.ts

```diff
@@ -37,8 +38,9 @@ export class VariantParserService {
       /**For each optionType iterate over each variant in varaints */
       variants.forEach(variant => {
         /**For option values like [small, Red] etc in varaint iterate over each option value */
-        this.currVariantOptionValues = variant.option_values;
-        variant.option_values.forEach(optionValue => {
+        this.currVariantOptionValues = variant.options;
+
+        variant.options.forEach(optionValue => {  
           /**
           * This loop runs for 750 times for 2 optiontypes and optionsvalues 3 and 5
```

Comment:
loop runs for 750 times :confused:

Reply:
Look's like we are grouping based on `option type`. We can loop over `option values` and then reduce them based on `option type`
https://www.consolelog.io/group-by-in-javascript/